### DISCUSSION TITLE
Bookmark file for syncing

### DISCUSSION THREAD

**Discussion by trymeouteh** [PaulWoitaschek/Voice#1346] [Ideas]
I use Syncthing to sync my audiobook between my computer and mobile devices. It would be nice for the app to have an option to store a simple plain text file which can be made for each audiobook and save the position on where you left off.

Obviously we need a open standard for this to work with other clients and I opened up the same request with the cozy app.

A directory should be chosen on where these text files are stored so they can be in a separate place than the audiobooks are stored.

**Comment by trymeouteh** [comment]
https://github.com/geigi/cozy/issues/542

**Comment by gamingexpert13** [comment]
Smart Audiobook Player (by Alex Kravchenko on the google play store) includes a position.sabp.dat file for the play/resume position in each audiobook's folder. You could maybe look at that file to see the structure he's using if you wanna make something standardized

**Reply by u6aab** [reply]
i installed smart audiobook player and listened until 2:32 in [this book](https://librivox.org/beowulf-by-unknown-2/), adding a bookmark at 2:21. it created 2 files in the folder where the book is, one called [bookmarks.sabp.xml](https://github.com/PaulWoitaschek/Voice/files/10421166/bookmarks.sabp.txt), and [position.sabp.dat](https://github.com/PaulWoitaschek/Voice/files/10421162/position.sabp.txt)

**Comment by kachapman** [comment]
Where are the bookmarks saved locally? I could just sync that folder with a cloud service.

**Comment by voidplayer** [comment]
> Where are the bookmarks saved locally? I could just sync that folder with a cloud service.

I couldnt find it either. It must be saved on a database

**Comment by AceKatz** [comment]
It's definitely in a database, just searched and found this:
https://github.com/PaulWoitaschek/Voice/blob/main/data/schemas/de.ph1b.audiobook.data.repo.internals.AppDb/45.json

**Comment by trymeouteh** [comment]
I would advise against saving the bookmark data inside the audio file since that file will need to be synced with other devices and audio files are not small in size especially full m4b audiobook files. It would be better to save the bookmark file into a separate file that is small in size.

**Reply by voidplayer** [reply]
> I would advise against saving the bookmark data inside the audio file since that file will need to be synced with other devices and audio files are not small in size especially full m4b audiobook files. It would be better to save the bookmark file into a separate file that is small in size.

It depends on how you use it. Temporary bookmarks, sure, dont save it to the file, but i use bookmarks like audiobooks use it. To define chapters. I would very much for this info to go into the file so it works in any program that work with audiobooks
